tests/bootstrap.js

```diff
@@ -36,12 +34,19 @@ const waitForDb = async ({url, fn, retries=0}) => {
 const waitForCouch = async (url) => {
   await waitForDb({url: url, fn: async () => {
     try {
-      const {statusCode} = await curl.get(url);
-      if(statusCode === 200) {
+      const result = await isDBReady(url);
+
+      if (result.ready) {
```

Comment:
If I'm reading this correctly, `isDBReady` only resolves when `ready` is true, and otherwise it rejects. This means there's lots of dead code here.

Either change this code to assume `ready` is true when the promise resolves, _or_ change the `isDBReady` to always resolve and never reject.

I think I prefer the second option as it makes the handling code nicer - the server not being up yet is an expected case that we can handle by retrying. We only need the catch for unexpected things!

For example...

```js
const isDBReady = (dbUrl) => {
  const parsed = new url.URL(dbUrl);
  const http = parsed.protocol === 'https:' ? require('https') : require('http');

  return new Promise((resolve, reject) => {
    http
      .get(dbUrl, res => {
        resolve({ ready: res.statusCode === 200 });
      })
      .on('error', (error) => {
        resolve({ ready: false, error });
      });
  });
};
```

Reply:
Ah much better, Done!